apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts

```diff
@@ -47,10 +54,77 @@ import { HttpRequest } from '../../app.types'
 export class PublicFlightController {
   constructor(
     private readonly flightService: FlightService,
+    @Inject(forwardRef(() => DiscountService))
     private readonly discountService: DiscountService,
     private readonly nationalRegistryService: NationalRegistryService,
   ) {}
 
+  @Get('discounts/:discountCode/checkFlightStatus')
+  @ApiResponse({
+    status: 200,
+    description: 'Input flight is eligible for discount as a connection flight',
+  })
+  @ApiResponse({
+    status: 400,
+    description:
+      'User does not have any flights that may correspond to connection flight',
+  })
+  @ApiResponse({
+    status: 403,
+    description:
+      'The provided discount code is not intended for connecting flights',
+  })
+  @ApiResponse({ type: CheckFlightViewModel })
+  async checkFlightStatus(
+    @Param() params: CheckFlightParams,
+    @Body() flight: CheckFlightBody,
```

Comment:
When I try to use swagger I get TypeError: Window.fetch: HEAD or GET Request cannot have a body.
I assume because you are using a request body for the GET request, can we put this in query params instead?

Reply:
Moved everything into path parameters